### ISSUE TITLE
[v3] Default field group not working in array dialog

### ISSUE THREAD

**Issue by Aratramba** [sanity-io/sanity#3957]
**Describe the bug**

Default field groups don't work within a dialog opened from an array.

**To Reproduce**

- Create a schema with an array of objects. 
- The object has groups, one of which is set to `default: true`.
- See how this default is ignored ('all fields' group is shown).

```tsx
import {defineConfig} from 'sanity'
import {deskTool} from 'sanity/desk'

export default defineConfig({
  name: 'default',
  title: 'Sanity v3 test',
  projectId: 'xx',
  dataset: 'production',
  plugins: [deskTool()],
  schema: {
    types: [
      {
        name: 'test',
        title: 'Test',
        type: 'document',
        fields: [
          {
            name: 'arr',
            title: 'Array of objects',
            type: 'array',
            of: [
              {
                name: 'item',
                title: 'Item',
                type: 'object',
                groups: [
                  {
                    name: 'a',
                    title: 'A (default)',
                    default: true,
                  },
                  {
                    name: 'b',
                    title: 'B',
                  },
                  {
                    name: 'c',
                    title: 'C',
                  },
                ],
                fields: [
                  {
                    name: 'fieldA',
                    title: 'field A',
                    type: 'string',
                    group: ['a'],
                  },
                  {
                    name: 'fieldB',
                    title: 'field B',
                    type: 'string',
                    group: ['b'],
                  },
                  {
                    name: 'fieldC',
                    title: 'field C',
                    type: 'string',
                    group: ['c'],
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
})

```

**Expected behavior**

The default group should open, not the 'all fields' group.

**Screenshots**

❌ Group with name `A (default)` should be open when opening the modal:

![image](https://user-images.githubusercontent.com/580312/207320192-82752d3c-c0ce-4444-9f72-cb81926ea4ff.png)

✅ It works in the main form view:

![image](https://user-images.githubusercontent.com/580312/207320099-f04ea408-100a-4dfb-8258-5f7c0cdf359c.png)

✅ Also works in a nested object in the main form view:

![image](https://user-images.githubusercontent.com/580312/207320144-85b3f2de-1de2-4639-931f-1ac423f845d0.png)

**Which versions of Sanity are you using?**
sanity `3.0.6`

**Comment by Aratramba** [comment]
Any news on this? I see [a test case](https://github.com/sanity-io/sanity/commit/c2cf16c820bf05df81d072a6899cee724fa9c8c1) was recently added in 3.2.0 that covers array field groups. But it's missing a test for default groups.

In this screenshot 'Group 2 / group 1' should be open by default.

![image](https://user-images.githubusercontent.com/580312/214595939-0da6c8f4-3f59-4c7f-82ee-633c2dd7271d.png)

**Comment by Aratramba** [comment]
Still seeing this in `3.3.1`

**Comment by ninaandal** [comment]
Hi @Aratramba 👋 Thank you for reporting. This has been fixed in [v3.5.1](https://github.com/sanity-io/sanity/releases/tag/v3.5.1) 😊

**Comment by Aratramba** [comment]
Thanks for the fix, forms are looking cleaned up and beautiful again! 👍
